lib/rspec/core/memoized_helpers.rb

```diff
@@ -153,6 +153,17 @@ def initialize
 
         def fetch_or_store(key)
           @memoized.fetch(key) do # only first access pays for synchronization
+            if @mutex.locked? && !@mutex.owned?
```

Comment:
For 1.8.7 support, we have [our own Mutex](https://github.com/rspec/rspec-support/blob/5cc75bd37b1bcfb1b9c6c4835d1f1b74fb695fe3/lib/rspec/support/mutex.rb) (see comment for explanation of why), it'll need `locked?` to be defined, which you can get from [here](https://github.com/ruby/ruby/blob/v1_8_7_374/lib/thread.rb#L67-L72). And [our own ReentrantMutex](https://travis-ci.org/rspec/rspec-core/jobs/169634889#L444) which will need `locked?` and `owned?`, both of which should be easy to add. ([Travis confirms](https://travis-ci.org/rspec/rspec-core/jobs/169634889#L444)).